language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/ToValueGenerator.scala

```diff
@@ -78,10 +78,13 @@ object ToValueGenerator {
     damlType match {
       case TypeVar(tvName) =>
         CodeBlock.of("toValue$L.apply($L)", JavaEscaper.escapeString(tvName), accessor)
-      case TypePrim(
-          PrimTypeBool | PrimTypeInt64 | PrimTypeDecimal | PrimTypeText | PrimTypeParty,
-          _
-          ) =>
+      case TypeNumeric(_) =>
```

Comment:
Why does this need a special case? I think you should be able to add it to the case below with `| TypeNumeric(_)` without having to hardcode `Decimal` here. It is already taken care of in [this change](https://github.com/digital-asset/daml/pull/2666/files#diff-2f0e56bd045381487004c446bf4bd473R91).

Reply:
fixed